lib/test_prof/before_all.rb

```diff
@@ -16,26 +16,28 @@ class << self
       attr_accessor :adapter
 
       def begin_transaction
+        raise "begin_transaction does not support a block" if block_given?
         raise AdapterMissing if adapter.nil?
 
         config.run_hooks(:begin) do
           adapter.begin_transaction
         end
-        yield
-      end
-
-      def within_transaction
```

Comment:
Direct yield. I imagine it was used to indicate that something is done in a transaction, but that was actually misleading.